tests/unit/jwt-rsa.test.ts

```diff
@@ -203,6 +204,28 @@ describe("unit tests jwt verifier", () => {
         expect(statement).toThrow("Invalid signature");
         expect(statement).toThrow(JwtInvalidSignatureError);
       });
+      test("invalid signature - async", async () => {
+        const signedJwt = signJwt(
+          { kid: keypair.jwk.kid },
+          {},
+          keypair.privateKey,
+          false
+        );
+        const statement = () =>
+          verifyJwt(signedJwt, "https://example.com/path/to/jwks.json", {
+            audience: null,
+            issuer: null,
+          });
+        expect.assertions(2);
+        mockHttpsUri("https://example.com/path/to/jwks.json", {
+          responsePayload: JSON.stringify(keypair.jwks),
+        });
+        await expect(statement).rejects.toThrow("Invalid signature");
+        mockHttpsUri("https://example.com/path/to/jwks.json", {
+          responsePayload: JSON.stringify(keypair.jwks),
```

Comment:
nit: maybe rename `keypair.jwks` to something more descriptive

Reply:
Like what? To me `keypair.jwks` is descriptive but I'm wondering what your perspective is

Reply:
In that case is an invalid keypair right?

Reply:
`keypair` stands for an RSA key pair, a dummy one we use for tests.
Now it is in fact an object that has properties for the several 'manifestations' of the keypair:

- keypair.jwk --> the keypair in JWK form
- keypair.jwks --> the keypair in JWKS form (which is the JWK form wrapped in an object with key `keys`)
- keypair.publicKey --> the public key of the keypair
- keypair.privateKey --> the private key of the keypair

Etc etc

Reply:
I  just pushed a commit that adds inline docs to the `keypair` object so that when you hover over it in the IDE, you're explained what it is.